aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/strategy/ThrowableSerializationStrategy.java

```diff
@@ -32,4 +34,25 @@
      * @return a list of {@code ThrowableDescription}s which represent the provided {@code Throwable}
      */
     List<ThrowableDescription> describeInContext(Throwable throwable, List<Subsegment> subsegments);
+
+    /**
+     * Serializes a {@code Throwable} into a {@code ThrowableDescription}. Uses the provided subsegments to chain exceptions where
+     * possible.
+     *
+     * @param context
+     *            the {@link SegmentContext} for the current segment. May be null.
+     * @param throwable
+     *            the Throwable to serialize
+     * @param subsegments
+     *            the list of subsegment children in which to look for the same {@code Throwable} object, for chaining
+     *
+     * @return a list of {@code ThrowableDescription}s which represent the provided {@code Throwable}
+     */
+    default List<ThrowableDescription> describeInContext(
```

Comment:
Just to play devil's advocate since this is a fairly tedious change and new API, do we need the performance bump in the throwable serializer? It's only used for requests that are already "bad" by nature of having an exception that needs recording. 

I'm ok with leaving it, especially if you think the performance is still important here, but will leave it up to you.

Reply:
Actually, looking at it again, is the segment context (and this new API) necessary at all? Can't you just grab a subsegment out of the list in the original API, do `getCreator()` on it and use the `generateId` method there?

Reply:
I did that at first, but there's no guarantee that the list will contain any subsegments. In my simple demo service, we just have the one master segment. I could search the list as a fallback but I didn't think that was necessary since I'm hoping most users are using the default serializer, which is able to always provide the segment in use.

I also experimented with adding an `Entity` parameter. Maybe I should have kept that? It was definitely safer – the only way to reach this method in X-Ray's codebase is through `EntityImpl#addException`, and while extremely unlikely, there's no guarantee that the entity invoking `describeInContext` is the same entity that's stored by the segment context. This wasn't a problem with the `Entity` parameter because `EntityImpl` could just give `this` as an argument.

Reply:
Ok yeah I like passing the entity instead of the context in that case - if we're just using the context to retrieve the entity then it definitely sounds simpler and safer to pass in the entity.